### ISSUE TITLE
[Accessibility] Allow users to auto-accept all incoming Copilot Edits in Agent MOde

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#243357]
Type: <b>Feature Request</b>

CC @meganrogge 

When using Copilot Edits with agent mode, users need to approve or disapprove Copilot's code edits and/or command execution.

Of course, users need to pay a special attention to granting permission automatically; however, manual grant may cause some user burden (especially, against assistive technology users).

I suggest adding an option for users to choose whether or not they want to automatically grant the permission to Copilot agent. (Absolutely, it must be turned off by default and we need to add a warning against the auto-grant).

VS Code version: Code - Insiders 1.99.0-insider (c1fb4c3983bcc28e63dd159dbbb2ea98d352d33e, 2025-03-12T08:28:07.410Z)
OS version: Windows_NT x64 10.0.26100
Modes:

**Comment by meganrogge** [comment]
I don't think we can do this from a security perspective. Thoughts @roblourens ?

**Comment by roblourens** [comment]
I swear there's an issue in https://github.com/microsoft/vscode-copilot-release somewhere for this but I can't find it. Anyway it's something that we will look into for terminal commands, and probably for other tools like from MCP. It would definitely be user opt-in, with some way to distinguish safe and unsafe actions, but details tbd.

**Comment by meganrogge** [comment]
Cc @roblourens feedback on the unified view

**Comment by meganrogge** [comment]
@jooyoungseo there's now `chat.tools.autoApprove` from @connor4312, which does what you want here and controls whether tools should be automatically approved. Given that and the fact that there's a `Keep` all edits action, I think this can be closed.

**Comment by jooyoungseo** [comment]
@meganrogge Does `chat.tools.autoApprove` also auto-accept Copilot edits or just tool actions?

**Comment by meganrogge** [comment]
> Given that and the fact that there's a Keep all edits action, I think this can be closed.

@jooyoungseo , it just approves tool actions. 

Do you think that there should be a setting to accept all edits by default also?

**Comment by jooyoungseo** [comment]
@meganrogge Yes, I still think auto-approval for edits and terminal execution needs to be added as user opt-in setting.

**Comment by soundyogi** [comment]
any news on this? or are there workarounds?

**Comment by mci-kmd** [comment]
I use Git commits and/or staged change sets to guard against agent mode changes and the manual approval of these edits is just an unnecessary pain for me, so I hope to see this implemented.

**Comment by TweakStories** [comment]
Any news on this? Cursor has this already.

**Comment by jelenv** [comment]
yes, this change would be really nice. This behavior (auto approve of edits in agent mode) is the default in Amp Code and Claude Code too. It makes working with the agent more pleasant. 

We can already use the Edit mode if you want to manually approve edits done by the model. In agent mode this could be the default behavior.

**Comment by munkerench** [comment]
Please fix this or change it to work in some fashion other than the current debacle.  How do we move it to the top of the list?

NOTE: It may be time to admit letting the 12 year olds drive the aircraft carrier is not really a good idea...just sayin.

**Comment by meganrogge** [comment]
We now have `chat.tools.terminal.autoApprove` for the terminal
